batch2/batch/front_end/front_end.py

```diff
@@ -283,7 +285,22 @@
                 if 'memory' not in resources:
                     resources['memory'] = BATCH_JOB_DEFAULT_MEMORY
 
-                cores_mcpu = parse_cpu_in_mcpu(resources['cpu'])
+                req_cores_mcpu = parse_cpu_in_mcpu(resources['cpu'])
+                req_memory_bytes = parse_memory_in_bytes(resources['memory'])
+
+                if req_cores_mcpu == 0:
+                    raise web.HTTPBadRequest(
+                        reason=f'bad resource request for job {id}: '
```

Comment:
This needs to be "bad resource reqests" (note the s) or, the test for this messages needs to have the s removed.